polynote-frontend/polynote/comms.ts

```diff
@@ -117,7 +134,7 @@ export class SocketSession extends EventTarget {
         }
     }
 
-    addMessageListener(msgType: typeof Message, fn: ListenerCallback, removeWhenFalse: boolean = false) {
+    addMessageListener<M extends Message, C extends (new (...args: any[]) => M) & typeof Message>(msgType: C, fn: (...args: ConstructorParameters<typeof msgType>) => void, removeWhenFalse: boolean = false) {
```

Comment:
This change was so that we can statically verify that the callback handles the correct number/types of arguments for the given type. This helped me catch a lot of message changes that I'd missed (as well as a pre-existing bug!)

Reply:
Oooh nice!